camel/configs.py

```diff
@@ -128,3 +128,16 @@ def from_openai_function_list(
             functions=[func.as_dict() for func in function_list],
             function_call=function_call,
         )
+
+
+@dataclass(frozen=True, kw_only=True)
+class OpenSourceConfig(ChatGPTConfig):
+    model_path: str
+    server_url: str
+
+
+OPENAI_API_PARAMS = {param for param in asdict(ChatGPTConfig()).keys()}
```

Comment:
In general I do not like computations in the global scope. But let it hang out there for this PR.

Reply:
I guess we can directly make this a set of string literals? Because the set of all arguments to OpenAI API is fixed.